ferretdb_packaging/defineversion/docker_tags_test.go

```diff
@@ -0,0 +1,508 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"github.com/sethvargo/go-githubactions"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestDefineDockerTags(t *testing.T) {
+	for name, tc := range map[string]struct {
+		env      map[string]string
+		expected *images
+	}{
+		"pull_request": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "ferretdb",
+				"GITHUB_EVENT_NAME": "pull_request",
+				"GITHUB_HEAD_REF":   "docker-tag",
+				"GITHUB_REF_NAME":   "1/merge",
+				"GITHUB_REF_TYPE":   "branch",
+				"GITHUB_REPOSITORY": "FerretDB/documentdb",
+				"INPUT_PG_VERSION":  "16",
+			},
+			expected: &images{
+				developmentImages: []string{
+					"ghcr.io/ferretdb/postgres-documentdb-dev:pr-docker-tag",
+				},
+			},
+		},
+		"pull_request-other": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "ferretdb",
+				"GITHUB_EVENT_NAME": "pull_request",
+				"GITHUB_HEAD_REF":   "docker-tag",
+				"GITHUB_REF_NAME":   "1/merge",
+				"GITHUB_REF_TYPE":   "branch",
+				"GITHUB_REPOSITORY": "OtherOrg/OtherRepo",
+				"INPUT_PG_VERSION":  "16",
+			},
+			expected: &images{
+				developmentImages: []string{
+					"ghcr.io/otherorg/postgres-otherrepo-dev:pr-docker-tag",
+				},
+			},
+		},
+
+		"pull_request_target": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "ferretdb",
+				"GITHUB_EVENT_NAME": "pull_request_target",
+				"GITHUB_HEAD_REF":   "docker-tag",
+				"GITHUB_REF_NAME":   "ferretdb",
+				"GITHUB_REF_TYPE":   "branch",
+				"GITHUB_REPOSITORY": "FerretDB/documentdb",
+				"INPUT_PG_VERSION":  "16",
+			},
+			expected: &images{
+				developmentImages: []string{
+					"ghcr.io/ferretdb/postgres-documentdb-dev:pr-docker-tag",
+				},
+			},
+		},
+		"pull_request_target-other": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "ferretdb",
+				"GITHUB_EVENT_NAME": "pull_request_target",
+				"GITHUB_HEAD_REF":   "docker-tag",
+				"GITHUB_REF_NAME":   "ferretdb",
+				"GITHUB_REF_TYPE":   "branch",
+				"GITHUB_REPOSITORY": "OtherOrg/OtherRepo",
+				"INPUT_PG_VERSION":  "16",
+			},
+			expected: &images{
+				developmentImages: []string{
+					"ghcr.io/otherorg/postgres-otherrepo-dev:pr-docker-tag",
+				},
+			},
+		},
+
+		"push/ferretdb": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "",
+				"GITHUB_EVENT_NAME": "push",
+				"GITHUB_HEAD_REF":   "",
+				"GITHUB_REF_NAME":   "ferretdb",
+				"GITHUB_REF_TYPE":   "branch",
+				"GITHUB_REPOSITORY": "FerretDB/documentdb",
+				"INPUT_PG_VERSION":  "16",
+			},
+			expected: &images{
+				developmentImages: []string{
+					"ferretdb/postgres-documentdb-dev:ferretdb",
+					"ghcr.io/ferretdb/postgres-documentdb-dev:ferretdb",
+					"quay.io/ferretdb/postgres-documentdb-dev:ferretdb",
```

Comment:
The branch build for push to `ferretdb` branch, nightly cron `schedule` test case and manual trigger `workflow_run` test case all produce `ferretdb` tag for dev build